plugins/filter/merge.py

```diff
@@ -5,41 +5,59 @@
 from copy import deepcopy
 
 from ansible.errors import AnsibleFilterError
-from deepmerge import always_merger
 
 
 def merge_with_app(base, new):
     """Merge data from the "new" application to the "base" application"""
 
     if not isinstance(base, dict) or not isinstance(new, dict):
-        raise AnsibleFilterError("input apps definitions should be 'dict' types")
+        raise AnsibleFilterError(
+            "input apps definitions should be 'dict' types"
+        )
 
     if base.get("name") is None or new.get("name") is None:
-        raise AnsibleFilterError("input apps should have a name key")
+        raise AnsibleFilterError("input apps should have a 'name' key")
 
     if base["name"] != new["name"]:
         raise AnsibleFilterError("input apps should have the same name")
 
+    if not base["name"]:
+        raise AnsibleFilterError("input apps name cannot be empty")
+
     result = deepcopy(base)
 
     # Add or override services metadata
-    for service in result["services"]:
-        new_app_selected_services = filter(
-            lambda s: s.get("name") == service.get("name"), new["services"]
-        )
+    for base_service in result["services"]:
+        new_app_selected_services = [
+            s
+            for s in new["services"]
+            if s.get("name") == base_service.get("name")
+        ]
```

Comment:
why did you switch from filter to list comprehension?

Reply:
Pylint told me to do so. I was annoyed when he told me that it was "the old way to do things" :flushed: